.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,67 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main, release* ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ main ]
+  schedule:
+    - cron: '43 20 * * 4'
```

Comment:
Looks like once a week? Any particular reason for this time? Just seemed kinda exact with the 43 minutes.

Reply:
This was auto generated by github. I didn't pick the cron setting. Do you think it would be necessary more than 1 a week?

Reply:
Nah, it just seemed kinda oddly specific (20:43 on Tuesday) no problem with it.